### DISCUSSION TITLE
How Reality Work ?

### DISCUSSION THREAD

**Discussion by lambdaxyzt** [XTLS/Xray-core#1876] [Q&A]
hi did not find resource , how reality work ? you should send request to real IP address and GFW can't see that ?! 
and camouflage website should be foreign ? what happen exactly if it would be in country ?

**Comment by RPRX** [comment]
https://github.com/XTLS/REALITY/pull/2#issuecomment-1479956295

**Reply by lambdaxyzt** [reply]
so it is better to only send this Link :

title : Freedom to Attach ( an  anti censorship strategies ) :
link : https://zh.m.wikipedia.org/zh-tw/%E4%BE%9D%E9%99%84%E7%9A%84%E8%87%AA%E7%94%B1

:)

at the end your logic about goodness of reality protocol is below ? ( google tranlate it from chinese ) 

"Freedom to Attach is an anti-censorship tactic that attempts to make it financially impossible for censors to block content on the Internet. The author stores the sensitive content on the cloud service, because the cloud service is encrypted, the censors cannot selectively block e content on the cloud service, either allow access to the sensitive content, or block the cloud service Everything on . Because of the large number of companies using cloud services, censors generally consider cloud services "too important to stop". "

**Reply by RPRX** [reply]
你的英文，我看不懂

**Reply by lambdaxyzt** [reply]
so they filter content base on SNI , so why GFW just not compare Destination IP of packet and all possible ip addresses of SNI ( or just resolve SNI ) and if they does not match block all connection to destination IP or block all connection to destination IP with specific SNI address for example cloudflare IP ranges is obvious isn't it ?

**Reply by RPRX** [reply]
首先你的英文很不礼貌并且有大量的语法错误，你可以用中文。

其次，关于你的问题，说来话长，所以我不会回答。

**Reply by lambdaxyzt** [reply]
impolite ? !! , ok , I don't know chinese , and you said "it is a long story" do you think this reality will work for other countries and censorships ?

**Reply by RPRX** [reply]
It depends.

**Reply by RPRX** [reply]
At last, I suggest that you should spend some time learning how to speak English in a polite way,  sincerely.

**Reply by lambdaxyzt** [reply]
you said "it is long story !" , I assume I don't have some knowledge and you can't waste your time explaining to beginner , so I glad to help these things with your suggested resources
